Marlin/stepper.cpp

```diff
@@ -788,6 +804,13 @@ void Stepper::isr() {
 
   // If current block is finished, reset pointer
   if (all_steps_done) {
+    if(IS_CHUNK(current_block)) {
+      //sync planner position back up with stepper positions
+      Planner::sync_from_steppers();
+      //temperature ISR can get drowned out under high step rate, make sure it gets run
+      Temperature::isr();
```

Comment:
It's probably not 100% safe. We've already done some tweaking to leave space for the serial UART, and have improved the reliability of `Temperature::isr` generally, but if the stepper ISR is run at too high a rate, the temperature ISR does get missed. There is a flag indicating that the temperature ISR is active, so if the stepper ISR is interrupting it then that can be checked.